shared/src/econ/mod.rs

```diff
@@ -81,17 +81,17 @@ impl Zero for TokenAmount {
     }
 }
 
-impl PartialOrd for TokenAmount {
+impl Ord for TokenAmount {
     #[inline]
-    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
-        self.atto.partial_cmp(&other.atto)
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.atto.cmp(&other.atto)
     }
 }
 
-impl Ord for TokenAmount {
+impl PartialOrd for TokenAmount {
     #[inline]
-    fn cmp(&self, other: &Self) -> Ordering {
-        self.atto.cmp(&other.atto)
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
```

Comment:
Clippy was unhappy that `partial_cmp` could, in theory, return a different result than `cmp`. It wouldn't have, but this is the "correct" way to do it.

Reply:
heh, _"Not doing this may silently introduce an error upon refactoring."_, fair enough